File: Backend/src/models/Payment.ts
import { model, Schema, Document } from "mongoose";
import { UserDoc } from "./User.js";

export type PaymentDoc = {
  userId: Schema.Types.ObjectId | UserDoc;
  bankAccount: number;
  payment: string;
};

const paymentSchema = new Schema(
  {
    userId: {
      type: Schema.Types.ObjectId,
      ref: "User",
    },
    bankAccount: {
      type: Number,
      required: true,
    },
    payment: {
      type: String,
      required: true,
    },
  },
  { timestamps: true }
);

export const Payment = model("Payment", paymentSchema);

File: python/Backend/application/models/Payment.py
from datetime import datetime
from application.models.db import *


class Payment(Document):
    userId= ObjectIdField()
    bankAccount= IntField(required=True)
    payment=StringField(required=True)
    timestamps = DateTimeField(default=datetime.utcnow())
    def to_json(self):
        return {
            "userId": self.userId,
            "bankAccount": self.bankAccount,
            "payment": self.payment,
            "timestamps": self.timestamps,
        }